### PR TITLE
B 4 edit user

### PR DESCRIPTION
**Pull request by Brutt**

### CODE REVIEW

File: src/main/java/com/photostudio/web/servlet/user/UserServlet.java [hunk 1]
```diff
@@ -91,4 +92,23 @@ public void doDelete(HttpServletRequest request, HttpServletResponse response) {
...
\ No newline at end of file
+
+    public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String id = request.getParameter("id");
+        if (id == null) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            return;
+        }
+
+        ObjectMapper mapper = new ObjectMapper();
```
Comment (viktorterljuk) [inline src/main/java/com/photostudio/web/servlet/user/UserServlet.java]: Register in service locator and reuse than

File: src/main/webapp/WEB-INF/templates/add-user.html [hunk 2]
```diff
@@ -35,48 +35,29 @@ <h3 class="user__title-login">Login</h3>
...
-                <div class="form-group">
-                    <input type="text" class="form-control " placeholder="City" name="city">
-                </div>
-                <div class="form-group">
-                    <input type="text" class="form-control " placeholder="Address" name="address">
-                </div>
-                <div class="form-group">
-                    <input type="text" class="form-control " placeholder="ZIP code" name="zipCode">
-                </div>
+                <input type="text" class="form-control " placeholder="Title">
```
Comment (Yuliia-Sh) [inline src/main/webapp/WEB-INF/templates/add-user.html]: Для чего поле ввода без name? AdditionalInfo  - тоже так

**Review by Yuliia-Sh — DISMISSED** [review]

**Review by Yuliia-Sh — DISMISSED** [review]

**Review by ValentynaBondarenko — DISMISSED** [review]

File: src/main/java/com/photostudio/web/servlet/user/UserServlet.java [hunk 3]
```diff
@@ -16,6 +17,7 @@
 public class UserServlet extends HttpServlet {
     private UserService userService = ServiceLocator.getService(UserService.class);
     private CommonVariableAppendService commonVariableAppendService = new CommonVariableAppendService();
+    private ObjectMapper mapper = new ObjectMapper();
```
Comment (viktorterljuk) [inline src/main/java/com/photostudio/web/servlet/user/UserServlet.java:24]: We have ServiceLocator, why "new" then?

File: src/main/java/com/photostudio/web/servlet/user/UserServlet.java [hunk 4]
```diff
@@ -91,4 +93,22 @@ public void doDelete(HttpServletRequest request, HttpServletResponse response) {
...
+            return;
+        }
+
+        User user = mapper.readValue(request.getReader(), User.class);
+
+        try {
+            userService.edit(user);
+
+            response.setStatus(HttpServletResponse.SC_OK);
+        } catch (Exception e) {
```
Comment (viktorterljuk) [inline src/main/java/com/photostudio/web/servlet/user/UserServlet.java:120]: Why it is bad?
https://stackify.com/common-mistakes-handling-java-exception/

File: src/main/webapp/WEB-INF/templates/user-info.html [hunk 5]
```diff
@@ -34,55 +34,40 @@ <h1 class="page__title">Manage user</h1>
...
-
-                </div>
-            </div>
-            <div class="new__user-info">
-                <div class="form-group">
-                    <input type="text" class="form-control " readonly placeholder="Title" th:value="${user.title}">
+                    <input type="email" class="form-control  " name="email" id="email" placeholder="Email "
+                           th:value="${user.email}">
+                    <input type="text" class="form-control  " id="phoneNumber" placeholder="Phone"
+                           th:value="${user.phoneNumber}">
```
Comment (OleksandrGubariev) [inline src/main/webapp/WEB-INF/templates/user-info.html:40]: I think need add name for input with id phoneNumber

File: src/main/webapp/WEB-INF/templates/user-info.html [hunk 6]
```diff
@@ -34,55 +34,40 @@ <h1 class="page__title">Manage user</h1>
...
+                    <input type="text" class="form-control " id="city" placeholder="City" th:value="${user.city}">
+                    <input type="text" class="form-control " id="address" placeholder="Address"
+                           th:value="${user.address}">
+                    <input type="text" class="form-control " id="zip" placeholder="ZIP code"
+                           th:value="*{user.zip != 0}? *{user.zip} : ''">
+                    <input type="text" class="form-control form-control_info" id="additionalInfo"
                            placeholder="Additional Information" th:value="${user.additionalInfo}">
+                    <button class="btn btn__manage" th:onclick="'updateUser(\'' + ${user.getId()} + '\');'">Save
+                    </button>
                 </div>
```
Comment (OleksandrGubariev) [inline src/main/webapp/WEB-INF/templates/user-info.html:60]: In all inputs add name

**Review by OleksandrGubariev — APPROVED** [review]

**Review by viktorterljuk — APPROVED** [review]

**Pull request merged**